### PR TITLE
Add SM8650 debugcc entries

### PR DESCRIPTION
**Pull request by superna9999**
This adds Debugcc support for SM8650

### CODE REVIEW

File: sm8650.c [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+	.mux_mask = 0x3f,
+
+	.div_reg = 0x80f8,
+	.div_mask = 0xf,
+	.div_val = 3,
+};
+
+static struct measure_clk sm8650_clocks[] = {
+/* cam_cc_debug_mux is 0x6a */
+	{ "cam_cc_bps_ahb_clk", &gcc, 0x6A, &cam_cc, 0x17},
```
Comment (MarijnS95) [inline sm8650.c:128]: Nit: missing trailing spaces in all `cam_cc` entries:

```suggestion
	{ "cam_cc_bps_ahb_clk", &gcc, 0x6A, &cam_cc, 0x17 },
```

File: sm8650.c [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+static struct measure_clk sm8650_clocks[] = {
+	/* gcc entries */
+	/*
+	 * GCC Debug Muxes:
+	 * - apss_cc_debug_mux: 0x135
+	 * - cam_cc_debug_mux: 0x6A
+	 * - disp_cc_debug_mux: 0x6F
+	 * - gpu_cc_debug_mux: 0x18D
+	 * - video_cc_debug_mux: 0x78
+	 */
```
Comment (MarijnS95) [inline sm8650.c:135]: This kind of stuff will be more cleanly self-documented in #24.

File: sm8650.c [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+	{ "gcc_aggre_noc_pcie_axi_clk", &gcc, 0x3F },
```
Comment (MarijnS95) [inline sm8650.c:136]: Another nit: should we switch all these to lowercase hex?

**Review by MarijnS95 — APPROVED** [review]
SGTM (haven't checked any of the values) though perhaps we should land #24 and #26 first to make this even cleaner?

**Comment by superna9999** [comment]
@MarijnS95 sure, I'll rebase when both are merged

File: sm8650.c (hunk 3 above)
Comment (superna9999) [reply sm8650.c]: Done

File: sm8650.c (hunk 2 above)
Comment (superna9999) [reply sm8650.c]: Indeed, rebases on top of #24, will keep draft until merged

File: sm8650.c [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+static struct measure_clk sm8650_clocks[] = {
+	/* gcc entries */
+	/*
+	 * GCC Debug Muxes:
+	 * - apss_cc_debug_mux: 0x135
+	 * - cam_cc_debug_mux: 0x6a
+	 * - disp_cc_debug_mux: 0x6f
+	 * - gpu_cc_debug_mux: 0x18d
+	 * - video_cc_debug_mux: 0x78
+	 */
```
Comment (lumag) [inline sm8650.c:155]: We can probably drop this comment, it is self-documented in the data structures above.

File: sm8650.c [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+		.div_mask = 0xf,
+		.div_val = 2,
+	},
+
+	.xo_div4_reg = 0x62008,
+	.debug_ctl_reg = 0x62048,
+	.debug_status_reg = 0x6204c,
+};
+
+/* All leaf clocks are disabled for now untill we enable corresponding GDSCs */
```
Comment (lumag) [inline sm8650.c:65]: This is not correct anymore. You have enabled all the clocks.

File: sm8650.c [hunk 6]
```diff
@@ -0,0 +1,468 @@
...
+	{ "disp_cc_mdss_non_gdsc_ahb_clk", &disp_cc, 0x38 },
+	{ "disp_cc_mdss_pclk0_clk", &disp_cc, 0xF },
+	{ "disp_cc_mdss_pclk1_clk", &disp_cc, 0x10 },
+	{ "disp_cc_mdss_rscc_ahb_clk", &disp_cc, 0x3A },
+	{ "disp_cc_mdss_rscc_vsync_clk", &disp_cc, 0x39 },
+	{ "disp_cc_mdss_vsync1_clk", &disp_cc, 0x36 },
+	{ "disp_cc_mdss_vsync_clk", &disp_cc, 0x13 },
+	{ "disp_cc_sleep_clk", &disp_cc, 0x47 },
+	{ "disp_cc_xo_clk", &disp_cc, 0x45 },
+	{ "gpu_cc_ahb_clk", &gpu_cc, 0x18 },
```
Comment (lumag) [inline sm8650.c:434]: empty line or a comment please

**Review by lumag — CHANGES_REQUESTED** [review]
Mostly LGTM, few minor nits.

**Review by lumag — APPROVED** [review]

**Pull request merged**